src/apps/Permissions/Permissions.js

```diff
@@ -65,244 +35,177 @@ class Permissions extends React.Component {
-              >
-                <ScrollTopElement
-                  ref={el => {
-                    this._scrollTopElement = el
-                  }}
-                />
-
-                <Wrap>
-                  <Screen position={0} animate={animateScreens}>
-                    {location.screen === 'home' && (
-                      <Home
-                        apps={apps}
-                        appsLoading={appsLoading}
-                        permissionsLoading={permissionsLoading}
-                        onOpenApp={this.handleOpenApp}
-                        onOpenEntity={this.handleOpenEntity}
-                      />
-                    )}
-                  </Screen>
-
-                  <Screen position={1} animate={animateScreens}>
-                    {['app', 'entity'].includes(location.screen) && (
-                      <React.Fragment>
-                        {location.screen === 'app' && (
-                          <AppPermissions
-                            app={location.app}
-                            loading={appsLoading}
-                            address={location.address}
-                            onManageRole={this.handleManageRole}
-                          />
-                        )}
-                        {location.screen === 'entity' && (
-                          <EntityPermissions
-                            title="Permissions granted to this entity"
-                            loading={appsLoading || permissionsLoading}
-                            address={location.address}
-                          />
-                        )}
-                      </React.Fragment>
-                    )}
-                  </Screen>
-                </Wrap>
-              </AppLayout>
-
-              <AssignPermissionPanel
-                apps={apps}
-                opened={showAssignPermissionPanel}
-                onClose={this.closeAssignPermissionPanel}
-                wrapper={wrapper}
-              />
-
-              <ManageRolePanel
-                app={location.app}
-                apps={apps}
-                opened={managedRole !== null}
-                onClose={this.closeManageRolePanel}
-                role={managedRole}
+  }, [apps, params, onParamsRequest])
+
+  const location = getLocation(params, apps)
+
+  const managedRole =
+    location.screen === 'app' &&
+    location.app &&
+    location.secondaryScreen === 'role'
+      ? resolveRole(location.app.proxyAddress, location.secondaryData)
+      : null
+
+  return (
+    <React.Fragment>
+      <div
+        css={`
+          // This element is only used to reset
+          // the view scroll using scrollIntoView()
+          position: absolute;
+          top: 0;
+          left: 0;
+          width: 1px;
+          height: 1px;
+        `}
+        ref={scrollTopElement}
+      />
+
+      <Header
+        primary={
+          location.screen === 'app' && location.app ? (
+            <Header.Title>
```

Comment:
We might need to update how this renders for mobile/smaller viewports with very long custom labels?

![image](https://user-images.githubusercontent.com/3072458/63257410-00fead80-c27a-11e9-92f9-2b1cad29f9d9.png)

Reply:
Yes! Fixed:

<img src=https://user-images.githubusercontent.com/36158/63390985-34902380-c3b1-11e9-8f41-0d61ec9a0d6f.png width=50%>